### ISSUE TITLE
Add option to not track completion of reengagement

### ISSUE THREAD

**Issue by austin-powell** [catalyst/moodle-mod_reengagement#45]
Based on my understanding of how this code works, I'm not sure that this is possible without a complete rewrite but I'll ask anyway.

Is there a reason why reengagements are forced to have activity completion turned on for them with "Show activity as complete when conditions are met"?

This causes all reengagements to get counted by Moodle's default get_course_progress_percentage function that is used (among other places) in the updated course overview block for Moodle 3.3 and on. The reason why this is an issue is that we for example have an reengagement set to occur 5 minutes after enrollment that sends a welcome email to these students. But because this reengagement then gets counted as "complete" users who have never logged in still show 4% complete on anything using core Moodle's progress functions including our custom progress reports.

Is it possible to change this or am I correct in my understanding of the plugin and its coding?

**Comment by peterbulmer** [comment]
Hi Austin,
It's been a while since I looked at the relevant code, but my recollection is that it wouldn't require a complete rewrite, and 'show activity complete' would likely have been set under the assumption that it was harmless.
If anyone wants to look into this issue don't be discouraged - it might be a minor-effort thing.

**Comment by jeremy-schweitzer** [comment]
I agree that it would be helpful to have an option to set completion as none/not tracked.

**Comment by danmarsden** [comment]
Feel free to help us work on this! - pull requests and or funding are always welcome! :-)

**Comment by ccomincini** [comment]
Good morning, I believe that the option to avoid tracking the completion of the activity would be necessary, especially considering that many users complete the courses without completing the reengagement and that moodle dashboard does not report 100%  completion on the course because of the failed completion of the reengagement activity. 
This is a problem and causes many support tickets are opened by users who know they have completed the course and want to see 100% completion in the dashboard.

**Comment by Lukas-Sander** [comment]
After a lot of searching, I found the blatantly simple solution:
In mod_form.php, at line 186, the completion option is frozen after being set to COMPLETION_TRACKING_AUTOMATIC by default.
Simply comment out line 186 and you're good!

`
        $this->standard_coursemodule_elements();
        if ($mform->elementExists('completion')) {
            $mform->setDefault('completion', COMPLETION_TRACKING_AUTOMATIC);
            // $mform->freeze('completion');
        }
`

**Comment by Lukas-Sander** [comment]
After some digging and searching, I found the code lines that need to be commented out for you to be able to select a different completion tracking (e.g. not tracking).
You need to comment out the following lines in mod_form.php:

184-187: if ($mform->elementExists('completion')) { [...]
266: $toform->completion = COMPLETION_TRACKING_AUTOMATIC;
288: $fromform->completion = COMPLETION_TRACKING_AUTOMATIC;

**Comment by jarvil1** [comment]
Any progress on this fix?

**Comment by danmarsden** [comment]
@Urpokarhu1, feature development on our plugins typically requires someone to fund the time for our development team to implement the improvements. Feel free to reach out privately if this is something you are interested in or if you have internal development capabilities, pull requests are always very welcome!

**Comment by abautu** [comment]
I have faced this problem also (i.e. user completes all course modules but the course progress is missing completion the Reengagement module, and they will stay like that for the duration setting of the reengagement module).

This happens even if the "Suppress notification if target activity complete" is enabled.

I solved this issue with this https://github.com/catalyst/moodle-mod_reengagement/pull/169, that makes reengagement to become completed if the user already completed the "suppress" activity.
